index.jsx

```diff
@@ -247,15 +164,9 @@ export function middleware({ dispatch }) {
         }
 
         if (action.type === 'DEVICE_SELECTED') {
-            const { device } = action;
-            const serialports = Object.keys(device)
-                .filter(k => k.startsWith('serialport'))
-                .map(k => device[k]);
-
-            const serialport = pickSerialPort(serialports);
+            const serialport = action.device.serialPorts.slice(-1)[0];
```

Comment:
Perhaps this should be extracted to a function with a descriptive name, like you did for Link Monitor.